vlm/src/main/scala/LayoutTileSource.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import geotrellis.raster._
+import geotrellis.spark.tiling._
+import geotrellis.spark.{SpatialKey}
+
+/** Reads tiles by key from a [[RasterSource]] as keyed by a [[LayoutDefinition]]
+  * @note It is required that the [[RasterSource]] is pixel aligned with the [[LayoutDefinition]]
+  *
+  * @param source raster source that can be queried by bounding box
+  * @param layout definition of a tile grid over the pixel grid
+  */
+class LayoutTileSource(val source: RasterSource, val layout: LayoutDefinition) {
```

Comment:
Would it make sense to have a value or method that produces the keys which intersect the given `RasterSource`? That way the user knows which keys contain data and which ones don't.

Reply:
That sounds like it would be useful.

Reply:
`def` is better than `val` btw so we don't have to hold on to the keys collection longer than its useful, its cheap enough to regenerate.